src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -30,30 +30,44 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
   }
 
-  def createResource(resourceType: ResourceType, resourceId: ResourceId, policiesOpt: Option[Map[AccessPolicyName, AccessPolicyMembership]] = None, userInfo: UserInfo): Future[Resource] = {
-    accessPolicyDAO.createResource(Resource(resourceType.name, resourceId)) flatMap { resource =>
-      policiesOpt match {
-        case Some(policies) =>
-          Future.traverse(policies) { case (accessPolicyName, accessPolicyMembership) =>
-            val resourceAndPolicyName = ResourceAndPolicyName(Resource(resourceType.name, resourceId), accessPolicyName)
-            val loadedSubjects = Future.traverse(accessPolicyMembership.memberEmails) { directoryDAO.loadSubjectFromEmail }.map(subjOpts => subjOpts.collect { case Some(subj) => subj })
-            val roles = accessPolicyMembership.roles
-            val actions = accessPolicyMembership.actions
-
-            loadedSubjects.flatMap { subjects =>
-              createPolicy(resourceAndPolicyName, subjects, roles, actions)
-            }
-          }.map {_ => Resource(resourceType.name, resourceId)}
-
-        //if the user didn't specify policies to create, default to just creating the owner policy
-        case None =>
-          val role = resourceType.roles.find(_.roleName == resourceType.ownerRoleName).getOrElse(throw new WorkbenchException(s"owner role ${resourceType.ownerRoleName} does not exist in $resourceType"))
-          val subjects: Set[WorkbenchSubject] = Set(userInfo.userId)
-          val resourceAndPolicyName = ResourceAndPolicyName(resource, AccessPolicyName(role.roleName.value))
-
-          createPolicy(resourceAndPolicyName, subjects, Set(role.roleName), Set.empty).map {_ => Resource(resourceType.name, resourceId)}
+  /**
+    * Create a resource with default policies. The default policies contain 1 policy with the same name as the
+    * owner role for the resourceType, has the owner role, membership contains only userInfo
+    * @param resourceType
+    * @param resourceId
+    * @param userInfo
+    * @return
+    */
+  def createResource(resourceType: ResourceType, resourceId: ResourceId, userInfo: UserInfo): Future[Resource] = {
+    val ownerRole = resourceType.roles.find(_.roleName == resourceType.ownerRoleName).getOrElse(throw new WorkbenchException(s"owner role ${resourceType.ownerRoleName} does not exist in $resourceType"))
+    val defaultPolicies: Map[AccessPolicyName, AccessPolicyMembership] = Map(AccessPolicyName(ownerRole.roleName.value) -> AccessPolicyMembership(Set(userInfo.userEmail), Set.empty, Set(ownerRole.roleName)))
+    createResource(resourceType, resourceId, defaultPolicies, userInfo)
+  }
+
+  def createResource(resourceType: ResourceType, resourceId: ResourceId, policies: Map[AccessPolicyName, AccessPolicyMembership], userInfo: UserInfo): Future[Resource] = {
+    val ownerExists = policies.exists { case (_, membership) => membership.roles.contains(resourceType.ownerRoleName) && membership.memberEmails.nonEmpty }
```

Comment:
Should there also be a check in `overwritePolicy` that checks that the owner policy is not emptied?

Reply:
it would have to be a more sophisticated check: load all the policies and ensure at least one has owner role and has members